docs: :memo: clarify async components for Vue Router

TheDutchCoder opened the pull request:
## Description of Problem
The docs around async components can be a bit confusing when people use Vue Router, hopefully this change clarifies the use of async component with Vue Router a bit.

## Proposed Solution
- Change the example of `NextPage.vue` to a more generic `Modal.vue`
- Add a note about async components and Vue Router, with a link to the relevant section in the Router docs.

## Additional Information
Relates to issue: https://github.com/vuejs/docs-next/issues/1013

skirtles-code reviewed (changes requested):
Thanks for the PR!

We don't usually reference Vue Router in the main Vue docs but I agree that this is potentially confusing and warrants a mention.

Coming from Vue 2, I thought that async components and lazy loaded routes were effectively the same thing. I imagine a lot of people thought the same. My first reaction when I read this page was that it would be a hassle to upgrade my router config. Hopefully the extra note can help others falling into the same trap.

TheDutchCoder commented on src/guide/migration/async-components.md:
````diff
@@ -45,25 +45,29 @@ import ErrorComponent from './components/ErrorComponent.vue'
 ```
 
+::: tip NOTE
+When using the official Vue Router and you want to asynchronously load (lazy load) router components, you should not use `defineAsyncComponent`. You can read more about this in the (Lazy Loading Routes)[https://next.router.vuejs.org/guide/advanced/lazy-loading.html#lazy-loading-routes] section of the Vue Router documentation.
````
Excellent, that's much better (whoops about the brackets lol, didn't even pick that one up!)

TheDutchCoder commented on src/guide/migration/async-components.md:
````diff
@@ -20,14 +20,14 @@ For a more in-depth explanation, read on!
 
 ```js
-const asyncPage = () => import('./NextPage.vue')
+const asyncPage = () => import('./CartFlyout.vue')
````
You're totally right, let me update that!

TheDutchCoder commented:
@skirtles-code I've updated the examples and also renamed `CartFlyout.vue` to a more illustrative `Modal.vue`, which I think is more familiar to people and makes more contextual sense.

Let me know what you think!

skirtles-code reviewed (approved):
Agreed. I prefer `Modal` too.
